inc/class-statify-dashboard.php

```diff
@@ -264,14 +264,18 @@ private static function _get_version() {
 	 * @since   0.1.0
 	 * @version 1.4.0
 	 *
-	 * @return  array  $data  Data from cache or DB
+	 * @param bool $force_refresh If true, the data is recalculated, otherwise the cached value is used if available.
+	 *
+	 * @return  array  $data  stats data from cache or database
 	 */
-	public static function get_stats() {
+	public static function get_stats( $force_refresh = false ) {
 
-		// Get from cache.
-		$data = get_transient( 'statify_data' );
-		if ( $data ) {
-			return $data;
+		// Get from cache if enabled.
+		if ( ! $force_refresh ) {
+			$data_from_cache = get_transient( 'statify_data' );
+			if ( $data_from_cache ) {
```

Comment:
Just a note:
This always causes a refresh if no data is stored in the database. `get_transient()` returns `false` if no value is available while the value is an empty string, if we've stored `null` before.

The behavior has been the same before, so nothing is changed here.
We _could_ think about either using strict comparison (`false === $data_from_cache`) here or - better imho - move the `set_transient()` call into the `(if ( ! empty( $data['visits'] ) ) { ... }` block and not store `null` at all.

Reply:
In think causing a refresh if no data is stored in the database is intended: For example, visiting the dashboard the first time after installing Statify should trigger getting the latest data from the database (if there are some already). That's why I called the parameter `$force_refresh` and not only `$refresh` (as a `$force_refresh = false` still might lead to a refresh).